loopy/tools.py

```diff
@@ -862,4 +865,40 @@ def t_unit_to_python(t_unit, var_name="t_unit",
     else:
         return python_code
 
+
+def memoize_on_disk(func, key_builder_t=LoopyKeyBuilder):
+    from loopy.version import DATA_MODEL_VERSION
+    from functools import wraps
+    from pytools.persistent_dict import WriteOncePersistentDict
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        from loopy import CACHING_ENABLED
+
+        if (not CACHING_ENABLED
+                or kwargs.pop("_no_memoize_on_disk", False)):
+            return func(*args, **kwargs)
+
+        transform_cache = WriteOncePersistentDict(
+            (f"loopy-memoize-cache-{key_builder_t.__qualname__}"
+             f"-{key_builder_t.__name__}-v0-{DATA_MODEL_VERSION}"),
```

Comment:
Put qualname and name next to each other to make this easier to read.

Reply:
Yep, agree.